app/src/org/commcare/views/EntityDetailView.java

```diff
@@ -434,4 +369,95 @@ private int getScreenWidth() {
         Display display = ((WindowManager)this.getContext().getSystemService(Context.WINDOW_SERVICE)).getDefaultDisplay();
         return display.getWidth();
     }
+
+    private void addSpinnerToGraph(WebView graphView, ViewGroup graphLayout) {
+        final ProgressBar spinner = new ProgressBar(this.getContext(), null, android.R.attr.progressBarStyleLarge);
+        spinner.setLayoutParams(new FrameLayout.LayoutParams(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT, Gravity.CENTER));
+        GraphLoader graphLoader = new GraphLoader((Activity) this.getContext(), spinner);
+
+        // Set up interface that JavaScript will call to hide the spinner
+        // once the graph has finished rendering.
+        ((WebView)graphView).addJavascriptInterface(graphLoader, "Android");

```

Comment:
> Would be good to get rid of the lint warning by either

Ended up doing both: skipping the method pre-jellybean, and suppressing the warning to make AS happy.